src/crimson/os/seastore/transaction_manager.h

```diff
@@ -174,25 +174,43 @@ class TransactionManager : public ExtentCallbackInterface {
   template <typename T>
   base_iertr::future<TCachedExtentRef<T>> read_pin(
     Transaction &t,
-    LBAMappingRef pin)
+    LBAMappingRef pin,
+    extent_len_t e_off,
+    extent_len_t e_len)
   {
     // checking the lba child must be atomic with creating
     // and linking the absent child
-    auto ret = get_extent_if_linked<T>(t, std::move(pin));
+    auto ret = get_extent_if_linked<T>(t, std::move(pin), e_off, e_len);
```

Comment:
I think maybe we can keep the old way of getting extents, and check whether they have the data we need, if not, load it from disk. What do you think?

Reply:
If we can confirm that every extent other than objectdatablock is fully loaded in memory, I think we can keep the old way and check whether the objectdatablock has the data in read path.

Reply:
Could you elaborate a little bit more about why the types of extents being read matter?

Reply:
Well. In current design, only ObjectDataBlock may be read partially. If we keep the old way of get_extent_if_linked, the extent will load from disk if it is not fully loaded(for lazy read design before), or it is fully loaded defaultly. Now if we keep the old way and do not send the e_off and e_len to this function, we should not load the data from disk in this function. Can we assume that only ObjectDataBlock may be not fully loaded? I need double check.

Reply:
I think you mean that `Cache::get_extent_viewable_by_trans()` always try to load non-fully-loaded extents, right?

If so, I think we can move the extent loading out to TransactionManager if non-logical-extents(lba/backref) are not supposed to be partial-loadable in the future, since all logical extents are loaded by TransactionManager. What do you think? 

CC @cyx1231st

Reply:
Seems to me it's reasonable to expose the partial read feature only through `TM::read_pin(t, pin, e_off, e_len)`, which is only used by `ObjectDataHandler`.